src/prophet/component.rs

```diff
@@ -257,6 +256,7 @@ impl AnnotationComponent {
         )
     }
 
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
```

Comment:
This constructor looks a little cursed, but I guess it was unavoidable.

Reply:
We could alternatively pull in the [derive_builder](https://crates.io/crates/derive_builder) crate if we wanted to go back and change some of the code and are ok with another dependency